cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PushCommand.java

```diff
@@ -0,0 +1,277 @@
+import io.ballerina.projects.util.ProjectUtils;
+import org.ballerinalang.central.client.CentralAPIClient;
+import org.ballerinalang.central.client.CentralClientException;
+import org.ballerinalang.central.client.NoPackageException;
+import org.ballerinalang.tool.BLauncherCmd;
+import org.ballerinalang.tool.LauncherUtils;
+import org.wso2.ballerinalang.util.RepoUtils;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.ballerina.cli.cmd.Constants.PUSH_COMMAND;
+import static io.ballerina.runtime.util.RuntimeConstants.SYSTEM_PROP_BAL_DEBUG;
+import static org.ballerinalang.tool.LauncherUtils.createLauncherException;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.IMPLEMENTATION_VERSION;
+import static org.wso2.ballerinalang.programfile.ProgramFileConstants.SUPPORTED_PLATFORMS;
+import static org.wso2.ballerinalang.util.RepoUtils.getRemoteRepoURL;
+
+/**
+ * This class represents the "ballerina push" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = PUSH_COMMAND, description = "push packages and binaries available locally to "
+        + "Ballerina Central")
+public class PushCommand implements BLauncherCmd {
+    private static PrintStream outStream = System.err;
+
+    @CommandLine.Parameters
+    private List<String> argList;
+
+    @CommandLine.Option(names = {"--help", "-h"}, hidden = true)
+    private boolean helpFlag;
+
+    @CommandLine.Option(names = "--debug", hidden = true)
+    private String debugPort;
+
+    @CommandLine.Option(names = "--repository", hidden = true)
+    private String repositoryHome;
+
+    @CommandLine.Option(names = {"--skip-source-check"}, description = "skip checking if source has changed")
+    private boolean skipSourceCheck;
+
+    @CommandLine.Option(names = "--experimental", description = "enable experimental language features")
+    private boolean experimentalFlag;
+
+    private Path userDir;
+    private PrintStream errStream;
+    
+    public PushCommand() {
+        userDir = Paths.get(System.getProperty(ProjectConstants.USER_DIR));
+        errStream = System.err;
+    }
+    
+    public PushCommand(Path userDir, PrintStream errStream) {
+        this.userDir = userDir;
+        this.errStream = errStream;
+    }
+
+    @Override
+    public void execute() {
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(PUSH_COMMAND);
+            outStream.println(commandUsageInfo);
+            return;
+        }
+
+        // Get source root path.
+        Path sourceRootPath = userDir;
+
+        // Push command only works inside a project
+        if (!ProjectUtils.isBallerinaProject(sourceRootPath)) {
```

Comment:
We have this validation inside the `Buildproject.load` method. We can remove this and handle the exception at L120.

Reply:
Done